ITISFoundation/osparc-simcore
# is1047/adapt simcore to ops

## What do these changes do?

- Defines the core stack ``${SWARM_STACK_NAME:-simcore}`` and an extra ``ops`` stack that includes a minimal amount of services from [ITISFoundation/osparc-ops](https://github/ITISFoundation/osparc-ops).
- ``minio`` was moved from the ``simcore`` stack to the ``tools`` stack. **NOTE**: this might require adapting deployer options in [ITISFoundation/osparc-ops](https://github/ITISFoundation/osparc-ops)
- ``make build*`` or ``make target=$(service) build*``: all or target image builds
```cmd
$ make help | grep build
build                Builds production images and tags them as 'local/{service-name}:production'. For single target e.g. 'make target=webserver build'
build-nc             As build but w/o cache (alias: rebuild)
build-devel          Builds development images and tags them as 'local/{service-name}:development'. For single target e.g. 'make target=webserver build-devel'
build-cache          Build cache images and tags them as 'local/{service-name}:cache'
```
- ``make up-*`` starts a swarm and deploys different stacks depending on the configuration
```cmd
$ make help | grep up
up-devel             Deploys local development stack, qx-compile+watch and ops stack (pass 'make ops_disabled=1 up-...' to disable)
up-prod              Deploys local production stack and ops stack (pass 'make ops_disabled=1 up-...' to disable)
up-version           Deploys versioned stack '$(DOCKER_REGISTRY)/{service}:$(DOCKER_IMAGE_TAG)' and ops stack (pass 'make ops_disabled=1 up-...' to disable)
```
- Removed tmp in makefile, instead it produces cocker-compose configs in place with prefix ``.``:
```
make .docker-compose-development.yml
make .docker-compose-production.yml
make .docker-compose-version.yml
```
NOTICE that these recipes do not need to be explicitly run since they are triggered via dependencies when calling ``up-*``.
- improved ``make info*``
```cmd
info                 displays setup information
info-images          lists tags and labels of built images. To display one: 'make name=webserver info-images'info-images
info-swarm           displays info about stacks and networks
```
- adapted fixtures in integration tests 
- MINOR: normalized default of usr/pass of all services
- Organized travis scripts: 
    - moved ``ops/travis`` to ``ci/travis``
    - under ``ci/travis`` ONLY scripts ( python tests moved to ``osparc-simcore/tests/swarm-deploy``
    - folders under ``ci/travis`` named after stages, therefore every script inside is a recipe (moved e2e scripts accordingly)

## Related issue number

#1047 

## How to test

       $ make help

To build and see images

       $ make info build build-devel
       $ make info-images

See images build for ``webserver``

       $ make name=webserver info-images
       $ make tag-latest
       $ make name=webserver info-images

Deploy ``simcore` and ``ops`` stack (latter is optional) and see swarm and its config

       $ make up-devel
       $ cat .docker-compose-development.yml
       $ make info-swarm

Same in production

       $ make down up-prod
       $ cat see .docker-compose-production.yml
       $ make info-swarm

Same w/o ``ops`` stack

       $ make ops_disabled=1 down up-prod


## Checklist
